src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/FormContext.cs

```diff
@@ -52,6 +51,19 @@ public IList<IHtmlContent> EndOfFormContent
 
         public bool CanRenderAtEndOfForm { get; set; }
 
+        private Dictionary<string, bool> RenderedFields
+        {
+            get
+            {
+                if (_renderedFields == null)
+                {
+                    _renderedFields = new Dictionary<string, bool>(StringComparer.Ordinal);
+                }
+
+                return _renderedFields;
+            }
+        }
```

Comment:
This does matter. The alternative is to write complicated code in `ViewContext` to try and make the formcontext lazy there. I explained this earlier.